koku/masu/processor/parquet/parquet_report_processor.py

```diff
@@ -582,7 +584,10 @@ def convert_csv_to_parquet(self, csv_filename: os.PathLike):  # noqa: C901
                 ),
                 exc_info=err,
             )
-            self.report_status.update_status(CombinedChoices.FAILED)
+            if self.report_status:
+                # internal masu endpoints may result in this being None,
+                # so guard this in case there is no status to update
+                self.report_status.update_status(CombinedChoices.FAILED)
```

Comment:
This looks like potentially two database calls, unless the result is cached as described above. If caching the value in an attribute isn't desireable, the walrus operator would keep this at one database call.
```suggestion
            if report_status := self.report_status:
                # internal masu endpoints may result in this being None,
                # so guard this in case there is no status to update
                report_status.update_status(CombinedChoices.FAILED)
```